File: Backend/src/Iit/Iit.Fibertest.Graph/DomainEntities/ExternalEvents/RtuAccident.cs
using Iit.Fibertest.Dto;

namespace Iit.Fibertest.Graph
{
    [Serializable]
    public class RtuAccident
    {
        public int Id { get; set; }
        public bool IsMeasurementProblem { get; set; } // Measurement vs Rtu
        public ReturnCode ReturnCode { get; set; }

        public DateTime EventRegistrationTimestamp { get; set; }
        public Guid RtuId { get; set; }
        public Guid TraceId { get; set; }
        public BaseRefType BaseRefType { get; set; }

        public int ClearedAccidentWithId { get; set; } 


        public string Comment { get; set; }

        public bool IsGoodAccident =>
            ReturnCode == ReturnCode.MeasurementEndedNormally 
            || ReturnCode == ReturnCode.MeasurementErrorCleared
            || ReturnCode == ReturnCode.MeasurementErrorClearedByInit
            || ReturnCode == ReturnCode.RtuManagerServiceWorking;
    }
}

File: WebFt30/src/app/core/store/models/ft30/rtu-accident.ts
import { BaseRefType } from './ft-enums';
import { ReturnCode } from './return-code';

export class RtuAccident {
  id!: number;
  isMeasurementProblem!: boolean;
  returnCode!: ReturnCode;

  registeredAt!: Date;
  rtuTitle!: string;
  rtuId!: string;
  traceTitle!: string;
  traceId!: string;
  baseRefType!: BaseRefType;

  comment!: string;
  clearedAccidentWithId!: number;
}
